projects/igniteui-angular/src/lib/grids/grid/grid.component.ts

```diff
@@ -1367,4 +1369,10 @@ export class IgxGridComponent extends IgxGridBaseDirective implements GridType,
         }
         return val;
     }
+
+    private shouldRegenerateColumns(oldData: any[] | null | undefined, newData: any[] | null | undefined): boolean {
```

Comment:
Could've add it as protected in the grid-base and reuse in the three grids instead if repeating it.